internal/pipe/docker/docker.go

```diff
@@ -328,3 +329,24 @@ func dockerPush(ctx *context.Context, image *artifact.Artifact) error {
 	ctx.Artifacts.Add(art)
 	return nil
 }
+
+func doPush(ctx *context.Context, img imager, name string, flags []string) (string, error) {
+	var try int
+	for try < 10 {
+		digest, err := img.Push(ctx, name, flags)
+		if err == nil {
+			return digest, nil
+		}
+		if strings.Contains(err.Error(), "received unexpected HTTP status: 503 Service Unavailable") {
```

Comment:
I believe all 5xx errors should be treated in that manner, is there any chance to check status code ?

Reply:
not really, that's just the output of `docker push`... but we can probably add a regex there